### ISSUE TITLE
Hidden rule in nn.FractionalMaxPool2d

### ISSUE THREAD

**Issue by PhyllisJi** [pytorch/pytorch#115531]
#### Description:

While using torch.nn.FractionalMaxPool2d, there is a hidden rule(seems to be
"kernel_size should be less than half of width and height of input tensor").
But this rule is not mentioned in the doc of FractionalMaxPool2d.

#### For example: 
```
x = torch.randn(3, 3, 7, 7)
p = torch.nn.FractionalMaxPool2d(kernel_size=4, output_ratio=0.8450560840300386, return_indices=False)
p(x)
```

this will tigger an error as below: 
```
----> 1 p(x)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\module.py:1501, in Module._call_impl(self, *args, **kwargs)
   1496 # If we don't have any hooks, we want to skip the rest of the logic in
   1497 # this function, and just call forward.
   1498 if not (self._backward_hooks or self._backward_pre_hooks or self._forward_hooks or self._forward_pre_hooks
   1499         or _global_backward_pre_hooks or _global_backward_hooks
   1500         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501     return forward_call(*args, **kwargs)
   1502 # Do not call functions when jit is used
   1503 full_backward_hooks, non_full_backward_hooks = [], []

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\modules\pooling.py:799, in FractionalMaxPool2d.forward(self, input)
    798 def forward(self, input: Tensor):
--> 799     return F.fractional_max_pool2d(
    800         input, self.kernel_size, self.output_size, self.output_ratio,
    801         self.return_indices,
    802         _random_samples=self._random_samples)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\_jit_internal.py:484, in boolean_dispatch.<locals>.fn(*args, **kwargs)
    482     return if_true(*args, **kwargs)
    483 else:
--> 484     return if_false(*args, **kwargs)

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\functional.py:503, in _fractional_max_pool2d(input, kernel_size, output_size, output_ratio, return_indices, _random_samples)
    492 if has_torch_function_variadic(input, _random_samples):
    493     return handle_torch_function(
    494         fractional_max_pool2d,
    495         (input, _random_samples),
   (...)
    501         _random_samples=_random_samples,
    502     )
--> 503 return fractional_max_pool2d_with_indices(
    504     input, kernel_size, output_size, output_ratio, return_indices, _random_samples
    505 )[0]

File ~\AppData\Local\Programs\Python\Python310\lib\site-packages\torch\nn\functional.py:482, in fractional_max_pool2d_with_indices(input, kernel_size, output_size, output_ratio, return_indices, _random_samples)
    480     n_batch = 1 if input.dim() == 3 else input.size(0)
    481     _random_samples = torch.rand(n_batch, input.size(-3), 2, dtype=input.dtype, device=input.device)
--> 482 return torch._C._nn.fractional_max_pool2d(input, kernel_size, output_size, _random_samples)

RuntimeError: fractional_max_pool2d(): pool height 4 too large relative to input height 7
```

### Suggest a potential alternative/fix

This should be supplemented in the documentation

cc @svekars @carljparker @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by mikaylagawarecki** [comment]
We would accept a PR that updates the documentation to include this constraint

**Comment by mikaylagawarecki** [comment]
Fixed by #116261
